x-pack/solutions/search/plugins/enterprise_search/server/deprecations/index.ts

```diff
@@ -0,0 +1,366 @@
+            'Therefore, Elastic Crawler is not supported in versions >= 9.x. ' +
+            'In order to upgrade other Native Connectors, metadata records in the `.elastic-connectors` index specific to ' +
+            'Elastic Crawler must be removed. For full details, see the documentation.',
+        }),
+        documentationUrl:
+          'https://elastic.co/guide/en/enterprise-search/current/upgrading-to-9-x.html',
+        correctiveActions: {
+          manualSteps: [
+            i18n.translate('xpack.enterpriseSearch.deprecations.crawler.listConnectors', {
+              defaultMessage: 'Enumerate all connector records',
+            }),
+            i18n.translate('xpack.enterpriseSearch.deprecations.crawler.deleteCrawlers', {
+              defaultMessage: 'Delete any that have `service_type: elastic-crawler`',
+            }),
+          ],
+          api: {
+            method: 'POST',
+            path: '/internal/enterprise_search/deprecations/delete_crawler_connectors',
+            body: {
+              ids: crawlers.map((it) => it.id),
+            },
+          },
+        },
+      },
+    ];
+  }
+}
+
+/**
+ * if the customer is using Native Connectors, agentless is available, but the integration server is missing, they are told that Integrations Server must be added
+ * if the customer is using Native Connectors, and agentless is unavailable, they are told that they must convert their connectors to Connector Clients
+ * if the customer was using "native" connectors that don't match our connector service types, they must delete them or convert them to connector clients.
+ */
+export async function getNativeConnectorDeprecations(
+  ctx: GetDeprecationsContext,
+  hasAgentless: boolean,
+  hasFleetServer: boolean
+): Promise<DeprecationsDetails[]> {
+  const client = ctx.esClient.asInternalUser;
+  const connectors: Connector[] = await fetchConnectors(client, undefined, false, undefined);
+  const nativeConnectors = connectors.filter((hit) => hit.is_native);
+  if (nativeConnectors.length === 0) {
+    return []; // no deprecations to register if there are no Native Connectors
+  } else {
+    const deprecations: DeprecationsDetails[] = [];
+
+    if (nativeConnectors.length > 0 && !hasAgentless) {
+      // you can't have elastic-managed connectors in an environment that Elastic doesn't manage
+      // ... and agentless is available in every Elastic-managed environment
+
+      deprecations.push({
+        level: 'critical',
+        deprecationType: 'feature',
+        title: i18n.translate('xpack.enterpriseSearch.deprecations.notManaged.title', {
+          defaultMessage:
+            'Connectors with `is_native: true` are only allowed in Elastic-managed environments',
+        }),
+        message: i18n.translate('xpack.enterpriseSearch.deprecations.notManaged.message', {
+          defaultMessage:
+            '"Native Connectors" are only intended for Elastic-managed environments like Elastic Cloud and ' +
```

Comment:
```suggestion
            '"Native Connectors" are managed services in Elastic-managed environments such as Elastic Cloud and ' +
```

Reply:
Elastic Cloud includes Serverless.
We should either use `Elastic Cloud Hosted and Elastic Serverless` or just `Elastic Cloud`.

Reply:
good catch, fixed to use the former